### PR TITLE
OSSM-6579: SM 3.0 Document RevisionBased update workflow

### PR DESCRIPTION
**Pull request by rh-tokeefe**
Version(s): Service Mesh 3.0

Issue: 
https://issues.redhat.com/browse/OSSM-6579

Link to docs preview:
https://76124--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/update/ossm-updating-openshift-service-mesh.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ossm-about-revisionbased-strategy.adoc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+:_mod-docs-content-type: Concept
+[id="about-revisionbased-strategy"]
+= About RevisionBased strategy
+:context: ossm-about-revisionbased-strategy
+
```
Comment (rh-tokeefe) [inline modules/ossm-about-revisionbased-strategy.adoc:5]: During updates, the `RevisionBased` strategy typically runs two revisions of a control plane. This capability allows you to gradually migrate workloads from the old control plane to the new control plane, making canary upgrades possible. This update strategy also allows you to upgrade by more than one minor version.

File: modules/ossm-performing-revisionbased-update.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
+:_mod-docs-content-type: Procedure
+[id="performing-revisionbased-update"]
+= Performing RevisionBased update
+:context: ossm-performing-revisionbased-update
+
+When updating {istio} using the `RvisionBased` strategy, you increment by more than one minor version at a time.
```
Comment (rh-tokeefe) [inline modules/ossm-performing-revisionbased-update.adoc:6]:
```suggestion
When updating {istio} using the `RevisionBased` strategy, you may increment by more than one minor version at a time. 
```

File: modules/ossm-performing-revisionbased-update.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+[source,yaml]
+----
+kind: Istio
+spec:
+  version: 1.20.2
+  updateStrategy: 
+    type: RevisionBased
+----
++
+The Sail operator deploys a new version of the control plane alongside the old version of the control plane. The sidecars remain connected to the old control plane.
```
Comment (rh-tokeefe) [inline modules/ossm-performing-revisionbased-update.adoc:25]:
```suggestion
The OSSM operator deploys a new version of the control plane alongside the old version of the control plane. The sidecars remain connected to the old control plane.
```

File: modules/ossm-performing-revisionbased-update.adoc (hunk 2 above)
Comment (rh-tokeefe) [reply modules/ossm-performing-revisionbased-update.adoc:6]: done

File: modules/ossm-performing-revisionbased-update.adoc (hunk 3 above)
Comment (rh-tokeefe) [reply modules/ossm-performing-revisionbased-update.adoc:25]: done

File: modules/ossm-about-deployment-and-update-strategies.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+* `InPlace`
+* `RevisionBased`
+
+The default strategy is the `InPlace` strategy.
+
+For more information, see the following documentation:
+
+* "About InPlace strategy"
+* "About RevisionBased strategy"
```
Comment (FilipB) [inline modules/ossm-about-deployment-and-update-strategies.adoc:16]: I think this should be moved to this section https://75532--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/install/ossm-installing-openshift-service-mesh.html#ossm-about-deployment-and-update-strategies

File: modules/ossm-performing-revisionbased-update.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
+:_mod-docs-content-type: Procedure
+[id="performing-revisionbased-update"]
+= Performing RevisionBased update
+:context: ossm-performing-revisionbased-update
+
+When updating {istio} using the `RvisionBased` strategy, you may increment by more than one minor version at a time because the existing sidecars do not automatically connect to the new revision of the control plane. Instead, you must manually move the workloads to the new control plane.
```
Comment (FilipB) [inline modules/ossm-performing-revisionbased-update.adoc:6]: Typo in RvisionBased

File: modules/ossm-about-revisionbased-strategy.adoc [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+Typically, an update that is performed using  the `RevisionBased` strategy runs two revisoins of a control plane. This capability allows you to gradually migrate workloads from the old control plane to the new control plane, making canary upgrades possible. The 'RevisionBased' strategy also allows you to update by more than one minor version.
```
Comment (FilipB) [inline modules/ossm-about-revisionbased-strategy.adoc:6]: Typo in `revisoins`

**Review by FilipB — CHANGES_REQUESTED** [review]

File: modules/ossm-performing-revisionbased-update.adoc (hunk 5 above)
Comment (rh-tokeefe) [reply modules/ossm-performing-revisionbased-update.adoc:6]: done

File: modules/ossm-about-revisionbased-strategy.adoc (hunk 6 above)
Comment (rh-tokeefe) [reply modules/ossm-about-revisionbased-strategy.adoc:6]: done

File: modules/ossm-about-deployment-and-update-strategies.adoc (hunk 4 above)
Comment (rh-tokeefe) [reply modules/ossm-about-deployment-and-update-strategies.adoc]: created OSSM-6632 to follow up on this

**Review by FilipB — APPROVED** [review]

File: modules/ossm-about-revisionbased-strategy.adoc [hunk 7]
```diff
@@ -0,0 +1,6 @@
+:_mod-docs-content-type: Concept
+[id="about-revisionbased-strategy"]
+= About RevisionBased strategy
+:context: ossm-about-revisionbased-strategy
+
+Typically, an update that is performed using  the `RevisionBased` strategy runs two revisions of a control plane. This capability allows you to gradually migrate workloads from the old control plane to the new control plane, making canary upgrades possible. The 'RevisionBased' strategy also allows you to update by more than one minor version.
```
Comment (gabriel-rh) [inline modules/ossm-about-revisionbased-strategy.adoc:6]: are the ticks wrong for the second occurrence of RevisionBased ?

**Review by gabriel-rh — APPROVED** [review]
one minor change but not essential

File: modules/ossm-about-revisionbased-strategy.adoc (hunk 7 above)
Comment (rh-tokeefe) [reply modules/ossm-about-revisionbased-strategy.adoc]: done

**Review by gabriel-rh — APPROVED** [review]
LGTM

**Pull request merged**